source-sftp/main.go

```diff
@@ -127,20 +135,39 @@ func (c config) PathRegex() string {
 }
 
 func newSftpSource(ctx context.Context, cfg config) (filesource.Store, error) {
+	var user = cfg.Credentials.Username
+	if len(user) == 0 && len(cfg.Username) > 0 {
+		user = cfg.Username
+	}
+
 	sshConfig := ssh.ClientConfig{
-		User:            cfg.Username,
+		User:            user,
 		Auth:            []ssh.AuthMethod{},
 		HostKeyCallback: ssh.InsecureIgnoreHostKey(),
 	}
 
-	// For now only password authentication is supported.
-	if cfg.Password != "" {
-		sshConfig.Auth = append(sshConfig.Auth, ssh.Password(cfg.Password))
-	} else if cfg.SSHKey != "" {
-		if signer, err := ssh.ParsePrivateKey([]byte(cfg.SSHKey)); err != nil {
-			return nil, fmt.Errorf("parsing ssh key: %w", err)
+	// Legacy authentication method
+	if cfg.Credentials.Type == "" {
+		if cfg.Password != "" {
```

Comment:
Do we have any tasks with the legacy authentication method that are using SSH key? I'd imagine not yet, and so the legacy authentication could be simplified to only consider the password case.